QRCoder/QRCodeGenerator.Polynom.cs

```diff
@@ -43,6 +154,134 @@ public override string ToString()
 
                 return sb.ToString();
             }
+
+            /// <inheritdoc/>
+            public void Dispose()
+            {
+                Free(_polyItems);
+                _polyItems = null;
+            }
+
+            /// <summary>
+            /// Ensures that the polynomial has enough capacity to store the specified number of polynomial terms.
+            /// </summary>
+            private void EnsureCapacity(int min)
+            {
+                if (_polyItems.Length < min)
+                {
+                    // All math by QRCoder should be done with fixed polynomials, so we don't need to grow the capacity.
+                    ThrowNotSupportedException();
+
+                    // Sample code for growing the capacity:
+                    //var newArray = Allocate(Math.Max(min - 1, 8) * 2); // Grow by 2x, but at least by 8
+                    //Array.Copy(_polyItems, newArray, _length);
+                    //Free(_polyItems);
+                    //_polyItems = newArray;
+                }
+
+                void ThrowNotSupportedException()
+                {
+                    throw new NotSupportedException("The polynomial capacity is fixed and cannot be increased.");
+                }
+            }
+
+#if NETCOREAPP
+            /// <summary>
+            /// Allocates memory for the polynomial terms.
+            /// </summary>
+            private static PolynomItem[] Allocate(int count)
```

Comment:
Nitpicking: the name `Allocate` is a bit misleading, as amortized no allocation will happen. Maybe call it `GetPolynomItems`, `Rent`, ...

Hm, but as in general memory allocation can also be thought of as renting from the virtual memory, maybe the name is still correct. In .NET-terms for allocation I think about that the GC hands me out a new created object...

Reply:
Done - RentArray & ReturnArray